File: cplus_a/Div.cpp
#include <vector>
#include <iostream>
#include <stdlib.h>

using std::cout;
using std::cin;
using std::endl;
using std::vector;
class Solution {
public:
    int divide(int dividend, int divisor) {
        if (dividend == 0 || divisor == 0) {
            return 0;
        }
        int i = 0;
        unsigned int ndivisor = divisor > 0 ? divisor : -1 * divisor;
        unsigned int ndividend = dividend > 0 ? dividend : -1 * dividend;

        while (ndividend >= ndivisor) {
                ndividend -= ndivisor;
                i++;     
        }
        if ((dividend > 0 && divisor > 0) || (dividend < 0 && divisor < 0)) {
            return int(i);
        } else {
            return -i;
        }
        
    }
};


int main () {
    Solution s;
    cout << int(s.divide(2147483648, -1)) << endl;
}

File: py_a/Div.py
class Solution(object):
    def divide(self, dividend, divisor):
        """
        :type dividend: int
        :type divisor: int
        :rtype: int
        """
        if dividend == 0 or divisor == 0:
            return 0


        ndivisor = abs(divisor)
        ndividend = abs(dividend)

        if ndivisor == 1:
            if divisor < 0 :
                return -dividend
            return dividend

        i = 0

        print(ndividend, ndivisor)
        while ndividend >= ndivisor:
            i += 1
            ndividend -= ndivisor

        if (divisor > 0 and dividend > 0) or (divisor < 0 and dividend < 0):
            return i
        return -i


if __name__ == "__main__":
    s = Solution()

    print(s.divide(-1, 1))
    print(s.divide(-1, -1))
    print(s.divide(-2147483648, -1))
  